### PR TITLE
Tee switch chain

### PR DESCRIPTION
**Pull request by conorbros**
This change allows you to switch the subchain used by Tee with a HTTP API.

In a follow up PR I want to add the ability to swap between returning responses from the main chain and the tee subchain

### CODE REVIEW

File: shotover-proxy/Cargo.toml [hunk 1]
```diff
@@ -53,6 +53,9 @@ opensearch = "2.1.0"
 serde_json = "1.0.103"
 time = { version = "0.3.25" }
 inferno = { version = "0.11.15", default-features = false, features = ["multithreaded", "nameattr"] }
+# workaround segfault in cassandra-cpp, seems to be caused by recent upgrade of openssl-src from 1.1.1 to 3.0.0
+openssl-sys = { version = "=0.9.92" }
```
Comment (conorbros) [inline shotover-proxy/Cargo.toml:57]: Oops, merge error I think

File: shotover/src/observability/mod.rs [hunk 2]
```diff
@@ -86,7 +86,7 @@ impl LogFilterHttpExporter {
                                                 format!("{:?}", error),
                                             )
                                         }
-                                        Ok(()) => rsp(StatusCode::NO_CONTENT, Body::empty()),
+                                        Ok(()) => rsp(StatusCode::OK, Body::empty()),
```
Comment (rukai) [inline shotover/src/observability/mod.rs:89]: what does this do?

**Comment by rukai** [comment]
Could you do a write up of how the user will use this new functionality and your reasoning behind the user facing parts?

From a quick look at this I'm thinking it would be simpler for the user to just have it always be Multi and just let them specify a single chain for the Single case.

File: shotover/src/observability/mod.rs (hunk 2 above)
Comment (conorbros) [reply shotover/src/observability/mod.rs:89]: I'll revert this, turns out NO_CONTENT was correct.

**Comment by conorbros** [comment]
After a bit of thought, I decided we should only have the ability to switch between the main and tee subchain, not the ability to switch between Tee subchains. 

The reasons being: I can't think of any good usecases and the potential for breakage with stateful connections.

I originally thought of it as a nice-to-have in addition to the main chain switching.

I'll pull those changes out and just have the main chain switching changes.

**Pull request closed without merging**